tesseract/tesseract_urdf/test/tesseract_urdf_mesh_unit.cpp

```diff
@@ -22,7 +22,7 @@ TEST(TesseractURDFUnit, parse_mesh)  // NOLINT
     EXPECT_FALSE(status->message().empty());
     EXPECT_TRUE(geom.size() == 1);
     EXPECT_TRUE(geom[0]->getTriangleCount() == 80);
-    EXPECT_TRUE(geom[0]->getVerticeCount() == 42);
+    EXPECT_TRUE(geom[0]->getVerticeCount() == 240);
```

Comment:
Why the vertices count change?

Reply:
I made changes to the assimp loader to include normals. I think that this is stopping vertices from being merged. Some vertices may have the same location, different faces may have different normals. This prevents the vertices from being merged. STL files include normals so I had to modify the unit tests to accept the higher vertices count.

I noticed a significant improvement in the quality of rendering when normals were included. Before, when normals were not specified, the rendered edges were very blurry and hard to see. With the normals, the edges are very clear. This video has an overview of normal map textures that shows how the normals are used to simulate depth: https://www.youtube.com/watch?v=6_-NNKc4lrk (I am not loading normal maps for our purposes, but the video shows how the shaders use the normals). I think that the renderer was doing an interpolation to generate normal vectors leading to bad edges.

Reply:
I assume you are referring to the edge of the triangles now are sharper? We need to make sure this only occurs for certain mesh types. This would be bad if it did this for convex_mesh types, if use the urdf version 2 it would cause problems because it does not automatically convert to convex hull. The default version does which is most likely why the tests are unaffected.

Reply:
Maybe we pass in a flag which indicates if it is for visual or collision. If for visual we process it this way and for collision we process it the previous way?

Reply:
As you suggested it looks like assimp does not perform this process `aiProcess_JoinIdenticalVertices` if normal's are requested to be parsed as outlined [here](`https://github.com/assimp/assimp/issues/407). I believe the previous approach should solve the issue. Then during processing at the urdf the parameters can be passed based on if it is a visual or collision geometry.

```
template <class T>
std::vector<std::shared_ptr<T>> createMeshFromPath(const std::string& path,
                                                   Eigen::Vector3d scale = Eigen::Vector3d(1, 1, 1),
                                                   bool triangulate = false,
                                                   bool flatten = false,
                                                   bool normals = false,
                                                   bool material_and_texture = false)
```

Reply:
Ok I will add the extra options

Reply:
This has been changed back to the old vertex count for collision only, and the new count when visual enabled. Tesseract itself doesn't use the visual so it shouldn't have any effect on the collision checkers and planners.